platforms/core-runtime/base-services/src/main/java/org/gradle/api/Named.java

```diff
@@ -15,9 +15,12 @@
  */
 package org.gradle.api;
 
+import org.gradle.declarative.dsl.model.annotations.HiddenInDefinition;
+
 /**
  * Types can implement this interface and use the embedded {@link Namer} implementation, to satisfy API that calls for a namer.
  */
+@HiddenInDefinition
```

Comment:
🤔 This is an interesting case.  It seems like its possible you might want to extend interfaces that are not Gradle-aware.  If so, you're only option would be to mark the extending type as `HiddenInDefinition` and then mark every property visible, which feels really awkward.  I wonder if we should be annotating these parent classes or if we should move this to the extending class, for instance, something like:
```
@HiddenInDefinition(Named.class) // Hide all elements coming from Named
interface Foo extends Named {
    ...
}
```

Reply:
Yep, I was thinking about something like that.
I think we can even add support for this use case with a separate annotation, so we don't add more responsibility to `HiddenInDefinition`.
And one more workaround for now is to introduce an intermediate supertype that inherits from the type you want to hide and mark the intermediate type as hidden.